pkg/common/utils.go

```diff
@@ -0,0 +1,21 @@
+package common
+
+import (
+	"math"
+	"math/rand"
+	"time"
+)
+
+// ExponentialBackoff is an algorithm which is used to spread out repeated execution of task (usually reconnection or packet
+// sending task to avoid network congestion or to decrease server load). In exponential backoff, wait time is increased
+// exponentially till threshold value. We have introduced jitter here to avoid thundering herd problem in future.
+func ExponentialBackoff(task func() error, baseTimeInMillis float64, multiplier float64, thresholdInMillis float64, retries int) error {
```

Comment:
Could we use an existing implementation like https://github.com/kubernetes/apimachinery/blob/master/pkg/util/wait/wait.go#L309

Reply:
changed the function to use existing implementation. It fitted nicely with the new restart listener approach.